rest-api-spec/src/main/resources/rest-api-spec/test/indices.upgrade/10_basic.yaml

```diff
@@ -9,14 +9,64 @@
             index:
               number_of_replicas: 0
 
+  - do:
+      indices.upgrade:
+        index: test_index
+
+  - match: {upgraded_indices.test_index.oldest_lucene_segment_version: '/(\d\.)+\d/'}
+  - is_true: upgraded_indices.test_index.upgrade_version
+
+---
+"Upgrade indices ignore unavailable":
 
   - do:
-      cluster.health:
-        wait_for_status: green
+      indices.create:
+        index: test_index
+        body:
+          settings:
+            index:
+              number_of_shards: 1
+              number_of_replicas: 0
 
   - do:
       indices.upgrade:
-        index: test_index
+        index: ["does_not_exist", "test_index"]
+        ignore_unavailable: true
 
-  - match: {upgraded_indices.test_index.oldest_lucene_segment_version: '/(\d\.)+\d/'}
+  - match: {_shards.total: 1}
   - is_true: upgraded_indices.test_index.upgrade_version
+  - is_false: upgraded_indices.test1
+
+---
+"Upgrade indices allow no indices":
+
+  - do:
+      indices.upgrade:
+        index: test_index
+        ignore_unavailable: true
+        allow_no_indices: true
+
+  - match: {_shards.total: 0}
+
+---
+"Upgrade indices disallow no indices":
+
+  - do:
+      catch: missing
```

Comment:
catch missing means that the status code must be 404. If we check the error message we don't check the status code anymore, the two checks are mutual exclusive. Shall we keep what we have?